lightning/src/util/events.rs

```diff
@@ -265,6 +323,12 @@ impl Writeable for Event {
 					(2, claim_from_onchain_tx, required),
 				});
 			},
+			&Event::ChannelClosed { ref channel_id, ref reason } => {
+				9u8.write(writer)?;
+				channel_id.write(writer)?;
+				reason.write(writer)?;
+				write_tlv_fields!(writer, {});
```

Comment:
Oops, sorry, all writes here need to be inside the TLV, not free-standing. This would allow for backwards compatibility, except that our `Ok(None)` read is also broken :(.

Reply:
Here, right!